### ISSUE TITLE
Error with TRAKTV.BUNDEL on Synology NAS ds213j

### ISSUE THREAD

**Issue by falconator** [trakt/Plex-Trakt-Scrobbler#277]
HI,

I'm having error in the logs, and TRAKTV won't work. Anybody can help me with that?
I'm installing it on a Synology NAS ds213j with a ARMv7 CPU.
here hare the TRAKTV and PLEX SERVER logs in attachment.

Thank you!

[com.plexapp.plugins.trakttv.log.txt](https://github.com/trakt/Plex-Trakt-Scrobbler/files/138343/com.plexapp.plugins.trakttv.log.txt)
[Plex Media Server.log.txt](https://github.com/trakt/Plex-Trakt-Scrobbler/files/138344/Plex.Media.Server.log.txt)

**Comment by fuzeman** [comment]
According to the [data sheet](https://global.download.synology.com/download/Document/DataSheet/DiskStation/13-year/DS213j/Synology_DS213j_Data_Sheet_enu.pdf) it has a "Marvell Kirkwood 1.2GHz" CPU, which appears to be ARMv5TE ([spec](https://origin-www.marvell.com/embedded-processors/kirkwood/)).

So you are in the same boat as #272, might have support for this platform available later this weekend.

**Comment by Saraz1** [comment]
Same bug ( even with 1.00.4 )

**Comment by gdolla** [comment]
Thanx so much everyone I didn't know what the issue was.  Your wonderful

**Comment by falconator** [comment]
Thank you for your response @fuzeman. Really apreciated that you take your time to help!
I know that on the Synology's Website it says that the CPU in the ds213j is a "Marvell Kirkwood 1.2GHz" but directly in my NAS in INFO CENTER it says that it is a "MARVELL Armada 370 88F6707" and it seems to be a ARMv7.
Who do we believe, the specs on the web or the SPECS in the NAS itself... confusing.

**Comment by Saraz1** [comment]
With plex, i download the armv7 version ( like now for the 0.15.6 )

**Comment by Noutbuk** [comment]
I do have a related problem and tried to investigate a little bit further (with my limited python skills).
I use a Synology ds215j to run Plex. It has a Marvell Armada 375, which is definitely an ARMv7 CPU. But still I can not successfully run the trakt.tv plugin. I attached the log of the trakt.tv plugin below.

I noticed that `Plug-ins/Trakttv.bundle/Contents/Libraries/Linux/armv5tel` is added to the library paths. This seems to be incorrect.
In the file `Trakttv.bundle\Contents\Libraries\Shared\plugin\core\libraries\helpers\system.py`, the arm float type is determined by looking at specific files. Both of them do not exist on my NAS. Maybe this is one of the problems.
```python
def arm_float_type(cls):
        if os.path.exists('/lib/arm-linux-gnueabihf'):
            return 'hf'
        if os.path.exists('/lib/arm-linux-gnueabi'):
            return 'sf'
        return None
```
I want to stress, that I am not good at python, but from the hints I would say, that the CPU architecture is not determined correctly.
If you need further information or help, I would be glad to provide them.

[com.plexapp.plugins.trakttv.txt](https://github.com/trakt/Plex-Trakt-Scrobbler/files/145156/com.plexapp.plugins.trakttv.txt)

**Comment by fuzeman** [comment]
@falconator @Noutbuk 
I've recently made some improvements to float-type detection in #275 that might resolve issues with some ARMv7 systems, try this version of the plugin: https://github.com/fuzeman/Plex-Trakt-Scrobbler/archive/feature/arm-float-detection.zip

**Comment by Saraz1** [comment]
Still not working for me " but still not showing in Channels " like in the other case
Plex 0.9.15.6 , Synology ds213 Dsm 6.0-7274

**Comment by Noutbuk** [comment]
Thanks for the work :)
Log seems to look better, but like @Saraz1 the plug-in is not showing in the channels tab.
Plex 0.9.15.6, Synology ds215j (Marvell Armada 375, ARMv7) DSM 5.2-5644 Update 5

Log looks similar to Saraz1´s: 
[com.plexapp.plugins.trakttv.txt](https://github.com/trakt/Plex-Trakt-Scrobbler/files/147933/com.plexapp.plugins.trakttv.txt)

**Comment by fuzeman** [comment]
Anyone with an ARMv7 Synology (Marvell Armada XP, 370, 375) can try the `feature/libraries/cpu/marvell-pj4` branch ([download](https://github.com/fuzeman/Plex-Trakt-Scrobbler/archive/feature/libraries/cpu/marvell-pj4.zip)) of the plugin.

I'm unable to test support for this device, so please post the latest plugin log file if the plugin doesn't startup correctly

**Comment by Saraz1** [comment]
Still not showing 

[com.plexapp.plugins.trakttv.txt](https://github.com/trakt/Plex-Trakt-Scrobbler/files/149820/com.plexapp.plugins.trakttv.txt)

**Comment by Noutbuk** [comment]
Same for me, still not showing.
If I can provide any further help, just let me know.

[com.plexapp.plugins.trakttv.txt](https://github.com/trakt/Plex-Trakt-Scrobbler/files/149822/com.plexapp.plugins.trakttv.txt)

**Comment by reimannje** [comment]
Hello!

Same here!

[com.plexapp.plugins.trakttv.log.txt](https://github.com/trakt/Plex-Trakt-Scrobbler/files/149824/com.plexapp.plugins.trakttv.log.txt)

**Comment by gdolla** [comment]
Hello My friends.  I removed that external drive I think it infected me
you soo much for all help!!

**Comment by fuzeman** [comment]
It looks like there might be some issues detecting the Synology devices and activating the correct libraries, so could you try the following changes:

@Saraz1 Could you try rename the ```marvell-pj4_64k``` folder to ```7-a``` in ```Plug-ins/Trakttv.bundle/Contents/Libraries/Linux/armv7_hf``` and then restart PMS.

@Noutbuk @reimannje Could you try rename the ```marvell-pj4_64k``` folder to ```marvell-pj4``` in ```Plug-ins/Trakttv.bundle/Contents/Libraries/Linux/armv7_hf``` and then restart PMS.


----------------

@Noutbuk Shouldn't need to remove that folder from ```Plug-in Support```, the plugin will update correctly if files are left in there. All the other steps look right though.

**Comment by Saraz1** [comment]
Its ok for the 213J ! 

I deleted all the trakt folder in my nas to have a fresh install , renamed the folder like u said and everything works fine. Thanks for your work and for your time.

**Comment by reimannje** [comment]
Hello!

Ist shows up in channels now but I get [Exception] Server Error - "server error" when trying to sync.

[com.plexapp.plugins.trakttv.log.txt](https://github.com/trakt/Plex-Trakt-Scrobbler/files/149864/com.plexapp.plugins.trakttv.log.txt)

**Comment by Noutbuk** [comment]
Works for me after renaming the folder, too. It shows up in the channels section and I successfully synced my collections with trakt. So no "Server Error" for me.
Thanks for the fast help and great work!

**Comment by reimannje** [comment]
Server Error is gone for me as well. 
All is good now!

Thank you so much for your help guys.

**Comment by fuzeman** [comment]
Great to hear that it's finally working after renaming that folder. Will make some changes *(so you don't need to rename that folder after every update)* and have a final fix available to test likely tomorrow.

--------------------------

@reimannje Those errors are returned from the Trakt.tv API, it's best to just wait awhile and try again when that happens.. hopefully Trakt will fix these issues soon..

--------------

@Saraz1 I still need to work out how to correctly identify the CPU on your system. The output of running `cat /proc/cpuinfo` in a terminal will be helpful if you have SSH access, otherwise a test release later today will display the required information in the log file.

> Btw , i have another question , about the filter , should i ask here or in the plex channels forum ? 

Noticed your question on the forum, will reply shortly :smiley:

**Comment by Saraz1** [comment]
/$ cat /proc/cpuinfo
Processor	: Marvell PJ4Bv7 Processor rev 1 (v7l)
BogoMIPS	: 1196.85
Features	: swp half thumb fastmult vfp edsp vfpv3 vfpv3d16 tls 
CPU implementer	: 0x56
CPU architecture: 7
CPU variant	: 0x1
CPU part	: 0x581
CPU revision	: 1

Hardware	: Marvell Armada-370
Revision	: 0000
Serial		: 0000000000000000

Btw, no more question , I found it myself ( to be honest, your faq helped me a lot ! ).

**Comment by fuzeman** [comment]
Could everyone test the latest changes on the `feature/libraries/cpu/marvell-pj4` branch, download it here: https://github.com/fuzeman/Plex-Trakt-Scrobbler/archive/feature/libraries/cpu/marvell-pj4.zip

You shouldn't need to rename any folders with this version, post your log files if there is any issues :smiley:

---------------------

@Saraz1 Thanks for that information, your system should be supported correctly with the latest changes.

> Btw, no more question , I found it myself ( to be honest, your faq helped me a lot ! ).

Woops, just remembered I hadn't answered yet. Glad the FAQ helped answer your question instead :smile:

**Comment by gdolla** [comment]
Hello smiles I removed that external hard drive I'm on a mac if that helps?

**Comment by fuzeman** [comment]
@gdolla Honestly, not sure which issue you want help with? You've replied to a number of issues recently with messages that don't seem to be related to the original issue report, so I'm a little confused at this point :confused: 

I'm guessing you are posting messages in the wrong issue? *(as this issue is related to Synology NAS, not Mac computers)*

**Comment by Saraz1** [comment]
That doesnt work with the new version . Im staying with the old one with the renamed folder
[com.plexapp.plugins.trakttv.txt](https://github.com/trakt/Plex-Trakt-Scrobbler/files/150647/com.plexapp.plugins.trakttv.txt)

**Comment by reimannje** [comment]
It works for me.

**Comment by Saraz1** [comment]
So there is a news , i downloaded the latest files you gives . Not working . 
I renamed the folder " marvell-pj4_64k " to " 7-a " like you ask with the old version . 
Restarted plex , and it works . So again , everythings works , even with the new one . I have just to rename the folder ( not a big deal for me )

**Comment by fuzeman** [comment]
@Saraz1 Interesting.. this appears to be the issue:
```
No valid processors returned from "/proc/cpuinfo"
```

Ah.. I know what the issue is, just need to make a few more changes to the cpu information parser.

> I renamed the folder " marvell-pj4_64k " to " 7-a " like you ask with the old version .
Restarted plex , and it works . So again , everythings works , even with the new one . I have just to rename the folder ( not a big deal for me )

I know renaming the folder isn't that hard, supporting your system without extra steps would be a lot cleaner though (for updates, and anyone else using a similar system).

--------

@reimannje Thanks for testing that, should have these changes included in a new release tomorrow.

**Comment by Saraz1** [comment]
Yeah , it would be better of course . But everythings works fine at the moment , so take your time . By the way , if you have any question , or test or something . Just ask .

**Comment by fuzeman** [comment]
@Saraz1 Pretty sure I've fixed the bug, try the latest version from here: https://github.com/fuzeman/Plex-Trakt-Scrobbler/archive/feature/libraries/cpu/marvell-pj4.zip

**Comment by Saraz1** [comment]
It works !!
No renamed folder needed . And it load way faster than before . It saved all my setup too .
Thanks for your time and your amazing work !

**Comment by fuzeman** [comment]
@Saraz1 Great :smiley: Looks like this is ready to release in a new version then (probably tomorrow).

**Comment by Noutbuk** [comment]
Just want to report that the newest version you posted also works for me. Thanks again for the fast feedback! Glad to have trakt on my NAS now.

**Comment by fuzeman** [comment]
Released in [v1.0.1.0](https://github.com/trakt/Plex-Trakt-Scrobbler/releases/tag/v1.0.1.0)
